tt_eager/tensor/tensor_impl.hpp

```diff
@@ -417,14 +417,14 @@ DeviceBuffer allocate_buffer_on_device(
 
 template <typename T>
 std::vector<T> read_data_from_device(const Tensor &tensor, uint32_t size_in_bytes) {
-    auto device_buffer = tensor.buffer();
+    auto device_buffer = tensor.device_buffer();
     TT_ASSERT(device_buffer->size() == size_in_bytes);
 
     const char *TT_METAL_SLOW_DISPATCH_MODE = std::getenv("TT_METAL_SLOW_DISPATCH_MODE");
     if (TT_METAL_SLOW_DISPATCH_MODE == nullptr) {
         std::vector<T> device_data;
         device_data.resize(size_in_bytes / sizeof(T));
-        EnqueueReadBuffer(tt::tt_metal::detail::GetCommandQueue(tensor.device()), *device_buffer, device_data.data(), true);
+        EnqueueReadBuffer(tt::tt_metal::detail::GetCommandQueue(tensor.device()), device_buffer, device_data.data(), true);
```

Comment:
Are you gonna change this API to receive which command queue to use after you add more software cq?

Reply:
Yes I'll add a new API that accepts the new sw CQ as a parameter